src/WebUI/Pods/PodsDetails.tsx

```diff
@@ -313,12 +313,25 @@ export class PodsDetails extends React.Component<IPodsDetailsProperties, IPodsDe
 
     // ToDO:: Handle GetImageDetails via ImageStore to avoid multiple calls to API from UI
     private _showImageDetails = (imageId: string) => {
-        const imageService = KubeFactory.getImageService();
-        imageService && imageService.getImageDetails(imageId).then(imageDetails => {
+        const showImageDetails = (imageDetails: IImageDetails): void => {
             this.setState({
                 selectedImageDetails: imageDetails
             });
-        });
+        }
+
+        let imageDetails: IImageDetails | undefined = this._imageDetailsStore.getImageDetails(imageId);
+        if (imageDetails) {
+            showImageDetails(imageDetails);
+        }
+        else {
+            const imageService = KubeFactory.getImageService();
+            imageService && imageService.getImageDetails(imageId).then(imageDetails => {
```

Comment:
variable name imageDetails  can be something like, fetchedImageDetails

Reply:
Changed accordingly